lib/widgets/sign_up_widget.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import '../configuration/app_text.dart';
+import '../screens/chat_rooms_screen.dart';
+import '../configuration/app_path.dart';
+import '../configuration/app_colors.dart';
+import '../services/auth.dart';
+
+class SignUp extends StatefulWidget {
+  @override
+  _SignUpState createState() => _SignUpState();
+}
+
+class _SignUpState extends State<SignUp> {
+  bool isLoading = false;
+
+  AuthMethods authMethods = new AuthMethods();
+
+  final formKey = GlobalKey<FormState>();
+  TextEditingController userNameTextEditingController =
+      new TextEditingController();
```

Comment:
new Dart versions have no 'new' keyword, you can create new objects without 'new' keyword. Delete it everywhere in the app